### DISCUSSION TITLE
Nullable support ? in F# - [RFC FS-1060] discussion/questions

### DISCUSSION THREAD

**Discussion by jkone27** [dotnet/fsharp#15619] [General]
question on [RFC FS-1060](https://github.com/dotnet/fsharp/pull/15181#top)

Q: just a small question what was the reason for adopting a `typscript like synthax` vs a `C# like` for nullable ref types and value types?  probably there is a valid reason for preferring a [typescript approach](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html) of `string | null`?  If so, would it be possible to type alias `string?` to `string | null` to make the 2 equivalent? 

```fsharp
let x : string? = null // C# approach

type string? = string nullable // Nullable<String> a possible definition of nullable?

//  as an "alternative" idea, this seems like a Higher order type in other languages 
// taking a type as parameter ?
type nullable<'a> = 'a | null 

// Make null the value of NullableOption.None ? then this could keep  NullableOption.Map and so on....
type NullOption =
    | Some of 'a
    |  null 
      
```

would that be a kind of anonymous discriminated union with "automatic" constructor? (like we have anonymous records),
would it work possibly for other "anonymous unions" ?

```fsharp

let x : string | null | int = 5 // anonymous DU
```

**Comment by kerams** [comment]
https://github.com/fsharp/fslang-design/blob/main/RFCs/FS-1060-nullable-reference-types.md#alternatives

**Comment by vzarytovskii** [comment]
Reasoning was that it can be more uniform with anonymous unions and potential future literal types. 

Aliasing `string?` is out of scope right now.
